cfn-resources/teams/cmd/resource/team-user/team_user.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package teamuser
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20231001001/admin"
+)
+
+type TeamUsersAPI interface {
+	GetUserByUsername(ctx context.Context, userName string) (*atlasv2.CloudAppUser, *http.Response, error)
+	AddTeamUser(ctx context.Context, orgID string, teamID string, addUserToTeam *[]atlasv2.AddUserToTeam) (*atlasv2.PaginatedApiAppUser, *http.Response, error)
+	RemoveTeamUser(ctx context.Context, orgID string, teamID string, userID string) (*http.Response, error)
+}
+
+type TeamUsersAPIService struct {
+	mongoDBCloudUsersAPI atlasv2.MongoDBCloudUsersApi
+	teamsAPI             atlasv2.TeamsApi
+}
+
+func NewTeamUsersAPIService(client *atlasv2.APIClient) *TeamUsersAPIService {
+	return &TeamUsersAPIService{
+		mongoDBCloudUsersAPI: client.MongoDBCloudUsersApi,
+		teamsAPI:             client.TeamsApi,
+	}
+}
+
+func (s *TeamUsersAPIService) GetUserByUsername(ctx context.Context, userName string) (*atlasv2.CloudAppUser, *http.Response, error) {
+	return s.mongoDBCloudUsersAPI.GetUserByUsername(context.Background(), userName).Execute()
+}
+
+func (s *TeamUsersAPIService) AddTeamUser(ctx context.Context, orgID string, teamID string, addUserToTeam *[]atlasv2.AddUserToTeam) (*atlasv2.PaginatedApiAppUser, *http.Response, error) {
+	return s.teamsAPI.AddTeamUser(ctx, orgID, teamID, addUserToTeam).Execute()
+}
+
+func (s *TeamUsersAPIService) RemoveTeamUser(ctx context.Context, orgID string, teamID string, userID string) (*http.Response, error) {
+	return s.teamsAPI.RemoveTeamUser(ctx, orgID, teamID, userID).Execute()
+}
+
+func FilterOnlyValidUsernames(mongoDBCloudUsersAPIClient TeamUsersAPI, usernames []string) ([]atlasv2.CloudAppUser, *http.Response, error) {
```

Comment:
[nit] define an interface with only the method you need
```go
type UserGetter interface {
  GetUserByUsername
}
```

Reply:
I think in team_user we need these three methods, so they are actually all needed. Are you suggesting one interface per method? I see that as a cons that will require more mocks and also more params passed to the methods.

Reply:
I moved these methods as private and provided an explanation below https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/810#discussion_r1391093913 - resolving this, we can continue in the other comment.